swarm/network/stream/streamer_test.go

```diff
@@ -921,3 +930,112 @@ func TestMaxPeerServersWithoutUnsubscribe(t *testing.T) {
 		}
 	}
 }
+
+/*
+TestGetSubscriptionsRPC sets up a simulation network of 16 nodes,
+starts the simulation, waits for SyncUpdateDelay in order to kick off
+stream registration, then tests that there are subscriptions.
+If provided with the `-printstats = true` option, it will print
+the information of who is subscribed to who to STDOUT
+*/
+func TestGetSubscriptionsRPC(t *testing.T) {
+	//arbitrarily set to 16
+	nodeCount := 16
+	//set the syncUpdateDelay for sync registrations to start
+	syncUpdateDelay := 500 * time.Millisecond
```

Comment:
Why delay _and_ healthy-check?

Reply:
The `syncUpdateDelay` is a streamer's internal delay time after which it starts actual syncing.
It is relevant to this test in so much as we need to wait at least this delay + the syncing duration time before we actually can check anything.

Check https://github.com/ethereum/go-ethereum/blob/c41721da4849f14ae5bb7fb4509ce94f3a155e53/swarm/network/stream/stream.go#L110-L112   

and

https://github.com/ethereum/go-ethereum/blob/c41721da4849f14ae5bb7fb4509ce94f3a155e53/swarm/network/stream/stream.go#L184

Reply:
Can't this also be done with messageevents?

Reply:
What do you mean? 

There is no "streamingStarted` event or something like that, and the `SyncUpdateDelay` is part of the main code, so we need to set it to something and wait for it.

Reply:
What I mean is then the streaming starts that streaming is actually devp2p messages being exchanged, no? Those messages could be inspected, I think;

* is it a syncSomethingType
* is it from a peer that hasn't sent this type of message yet

Not sure if I'm making myself clear...

Reply:
Yes I already updated the PR to wait for `SubscribeMsg`

Reply:
@nolash can you pls mark this as resolved if you agree or comment further if you think it still needs attention. Thanks

Reply:
> The syncUpdateDelay is a streamer's internal delay time after which it starts actual syncing.

In such cases we should always get the value directly from the source where it's defined IMO.

Reply:
@nolash in this case no.  If no `SyncUpdateDelay` is provided, the `streamer` will set it to `15 * time.Second`, which is obviously too long for a test.

Thus we need to set it, which is why it is set to `500 * time.Millisecond` for this test.

Check this https://github.com/ethereum/go-ethereum/pull/18284#discussion_r241424306 comment please, that has the relevant snippet